src/KeyVault/KeyVault/Models/KeyVaultCmdletBase.cs

```diff
@@ -85,6 +85,21 @@ protected string GetDefaultFileForOperation(string operationName, string vaultNa
             return filename;
         }
 
+        /// <summary>
+        /// Utility function that will split the Vault secret uri into 
+        /// `vaultName`, `secretName`, `Version`
+        /// </summary>
+        /// <param name="secretUri">The user's input unique resource identifier</param>
+        protected string[] SplitSecretUri(Uri secretUri)
```

Comment:
```suggestion
        protected string[] SplitSecretUri(Uri secretUri)
```
How about a new class named KeyVaultSecretUri, which contains property VaultName, Name, Version and a constructor with parameter string.